e2e/tests/session_steps_test.go

```diff
@@ -297,3 +250,80 @@ func (s *suite) waitForNewBlockEvent(
 		s.Log("Success; message detected before timeout.")
 	}
 }
+
+// newEventTypeMatchFn returns a function that matches an event based on its type
+// field. The type URL is constructed from the given module and eventType arguments
+// where module is the module name and eventType is the protobuf message type name
+// without the "Event" prefix; e.g., pass "tokenomics" and "ClaimSettled" to match
+// the "poktroll.tokenomics.EventClaimSettled" event.
+func newEventTypeMatchFn(module, eventType string) func(*abci.Event) bool {
+	targetEventType := fmt.Sprintf("poktroll.%s.Event%s", module, eventType)
+	return func(event *abci.Event) bool {
+		if event == nil {
+			return false
+		}
+
+		if event.Type == targetEventType {
+			return true
+		}
+		return false
+	}
+}
+
+// newEventMsgTypeMatchFn returns a function that matches an event based on the
+// "action" attribute in its attributes field, which is populated with the message
+// type URL of the message to which a given event corresponds. The target action
+// is constructed from the given module and msgType arguments where module is the
+// module name and msgType is the protobuf message type name without the "Msg" prefix;
+// e.g., pass "proof" and "CreateClaim" to match the "poktroll.proof.MsgCreateClaim" message.
+func newEventMsgTypeMatchFn(module, msgType string) func(event *abci.Event) bool {
+	targetMsgType := fmt.Sprintf("/poktroll.%s.Msg%s", module, msgType)
+	return func(event *abci.Event) bool {
+		if event == nil {
+			return false
+		}
+
+		for _, attribute := range event.Attributes {
+			if attribute.Key == "action" {
+				if attribute.Value == targetMsgType {
+					return true
+				}
+			}
+		}
+		return false
+	}
+}
+
+// newEventModeMatchFn returns a function that matches an event based on the
+// "mode" attribute in its attributes field. The target mode value is the given
+// mode string.
+func newEventModeMatchFn(mode string) func(event *abci.Event) bool {
```

Comment:
Maybe a `newEventAttributeMatchFn(key, value`) string that'll be used by both msgType and mode match function is worth it.